api/test/api-common.test.js

```diff
@@ -5554,6 +5608,151 @@ describe('API', () => {
+          expect(result.data.orders.length).toBe(20);
+          expect(result.data.nextPage).toBeDefined();
+        });
+
+        test('GET /datasetorders (isWorkerpoolStrict = true & workerpool = undefined): should return public orders including "any" workerpool', async () => {
+          const result = await request
+            .get(
+              buildQuery('/datasetorders', {
+                chainId, // *
+                dataset: datasetAddress, // *
+                isWorkerpoolStrict: true,
+              }),
+            )
+            .then(parseResult);
+          expect(result.status).toBe(OK_STATUS);
+          expect(result.data.ok).toBe(true);
+          expect(result.data.count).toBe(publicOrders.length);
+          expect(result.data.orders).toBeDefined();
+          expect(Array.isArray(result.data.orders)).toBe(true);
+          expect(result.data.orders.length).toBe(20);
+          expect(result.data.nextPage).toBeDefined();
+        });
+
+        test('GET /datasetorders (isRequesterStrict = true & requester = undefined): should return public orders including "any" requester', async () => {
+          const result = await request
+            .get(
+              buildQuery('/datasetorders', {
+                chainId, // *
+                dataset: datasetAddress, // *
+                isRequesterStrict: true,
+              }),
+            )
+            .then(parseResult);
+          expect(result.status).toBe(OK_STATUS);
+          expect(result.data.ok).toBe(true);
+          expect(result.data.count).toBe(publicOrders.length);
+          expect(result.data.orders).toBeDefined();
+          expect(Array.isArray(result.data.orders)).toBe(true);
+          expect(result.data.orders.length).toBe(20);
+          expect(result.data.nextPage).toBeDefined();
+        });
+
+        test('GET /datasetorders (app filter & isAppStrict): should exclude orders with "any" filtered app)', async () => {
+          const { data, status } = await request
+            .get(
+              buildQuery('/datasetorders', {
+                chainId, // *
+                dataset: datasetAddress, // *
+                app: allowedApp,
+                isAppStrict: true,
+              }),
+            )
+            .then(parseResult);
+
+          const ordersExcludingAnyApp = appAllowedOrders.filter(
+            (order) => order.order.apprestrict !== utils.NULL_ADDRESS,
+          );
+
+          expect(status).toBe(OK_STATUS);
+          expect(data.ok).toBe(true);
+          expect(data.count).toBe(ordersExcludingAnyApp.length);
+          expect(data.orders).toBeDefined();
+          expect(Array.isArray(data.orders)).toBe(true);
+          expect(data.orders.length).toBe(ordersExcludingAnyApp.length);
+          data.orders.forEach((e) => {
+            expect(
+              e.order.apprestrict === allowedApp ||
+                e.order.apprestrict !== utils.NULL_ADDRESS,
+            ).toBe(true);
```

Comment:
the check seems to be too permissive, we expect apprestrict to be allowedApp

```suggestion
            expect(e.order.apprestrict).toBe(allowedApp);
```

Reply:
commit : 6fab488418a00e8ef7004ae32cd22009218e443c